controller/applicationsController.js

```diff
@@ -0,0 +1,52 @@
+const Applications = require("../models/applications");
+const mongoose = require("mongoose");
+const CustomError = require("../utils/customError");
+const responseHandler = require("../utils/responseHandler");
+const Organizations = require("../models/organizations");
+
+exports.getSingleApplication = async (req, res, next) => {
+  try {
+    const { organizationId } = req.params;
+    const { applicationId } = req.params;
+
+    if (!mongoose.Types.ObjectId.isValid(applicationId)) {
+      return next(new CustomError(400, "Application's ID not found"));
+    }
+    if (!mongoose.Types.ObjectId.isValid(organizationId)) {
+      return next(new CustomError(400, "Organization's ID not found"));
+    }
+
+    const application = await Applications.findById(applicationId);
+    if (!application) {
+      return next(new CustomError(400, "Application not found"));
+    }
+    const organization = await Organizations.findById(organizationId);
+    if (!organization) {
+      return next(new CustomError(400, "Organization not found"));
+    }

```

Comment:
We are trying to find an application within an organization. You should be searching the application model only and testing if the returned `application.organizationId` is equal to `organizationId`.
Alterantively, do an `Application.find({_id: applicationId, organizationId: organizationId})` which will should return a matching application